### PR TITLE
flex button

### PR DESCRIPTION
**Pull request by giladgray**
#### Changes proposed in this pull request:

- `pt-button-base()` is inline flex row
- ⚠️  Button renders text and children inside `.pt-fill` container
  - vertical buttons automatically align left and push rightIcon to right side!
  - loading easily hides this container without needing to wrap children!
- Button now supports multiple lines of text, but will only wrap if forced to by layout constraints (like max-width)
- add `alignText` prop to `Button` and `ButtonGroup` (and examples)

#### Reviewers should focus on:

- new `align` prop: play with it in both `ButtonGroup` JS examples

### CODE REVIEW

File: packages/core/src/components/button/abstractButton.tsx [hunk 1]
```diff
@@ -22,6 +22,14 @@ export interface IButtonProps extends IActionProps {
      */
     active?: boolean;
 
+    /**
+     * Text alignment within button. By default, icons and text will be centered within the button.
+     * Passing this prop will cause the text container to fill the button and align the text within that
+     * to the appropriate side. `icon` and `rightIcon` will be pushed to either side.
+     * @default "center"
+     */
+    align?: "left" | "center" | "right";
```
Comment (llorca) [inline packages/core/src/components/button/abstractButton.tsx]: 😍 😍 😍

File: packages/core/src/components/button/buttonGroup.tsx [hunk 2]
```diff
@@ -41,13 +49,15 @@ export class ButtonGroup extends React.PureComponent<IButtonGroupProps, {}> {
...
+        const { align, className, fill, minimal, large, vertical, ...htmlProps } = this.props;
         const buttonGroupClasses = classNames(
             Classes.BUTTON_GROUP,
             {
+                [Classes.ALIGN_LEFT]: align === "left",
+                [Classes.ALIGN_RIGHT]: align === "right",
                 [Classes.FILL]: fill,
-                [Classes.MINIMAL]: minimal,
                 [Classes.LARGE]: large,
+                [Classes.MINIMAL]: minimal,
```
Comment (llorca) [inline packages/core/src/components/button/buttonGroup.tsx]: small?

File: packages/docs-app/src/examples/core-examples/buttonGroupExample.tsx [hunk 3]
```diff
@@ -8,37 +8,46 @@ import * as React from "react";
...
+    vertical: boolean;
 }
 
 export class ButtonGroupExample extends BaseExample<IButtonGroupExampleState> {
     public state: IButtonGroupExampleState = {
+        align: "center",
         fill: false,
+        iconOnly: false,
         large: false,
         minimal: false,
```
Comment (llorca) [inline packages/docs-app/src/examples/core-examples/buttonGroupExample.tsx]: small in this example would be neat

Comment (giladgray) [reply packages/docs-app/src/examples/core-examples/buttonGroupExample.tsx]: ButtonGroup doesn't have a `small` prop. 
let's wait on that, i want to add modifier props _everywhere_

File: packages/core/src/components/button/buttonGroup.tsx (hunk 2 above)
Comment (giladgray) [reply packages/core/src/components/button/buttonGroup.tsx]: would prefer not to add tangential new props in this PR.
i plan to add modifier props across the board after 2.0 (non-breaking)

File: packages/core/src/components/button/abstractButton.tsx (hunk 1 above)
Comment (adidahiya) [reply packages/core/src/components/button/abstractButton.tsx]: this conflicts somewhat with the `NavbarGroup` align prop -- here, you are referring to the alignment of _content_ while for NavbarGroup (as well as most design tools), `align` refers to the component's alignment in its container. I suggest renaming this to `textAlignment` or `alignText`

**Comment by adidahiya** [comment]
otherwise lgtm

File: packages/core/src/components/button/abstractButton.tsx (hunk 1 above)
Comment (giladgray) [reply packages/core/src/components/button/abstractButton.tsx]: mmm yes good catch @adidahiya! i'll go with `alignText`

**Pull request merged**